database/scrapers/departments.ts

```diff
@@ -0,0 +1,188 @@
+import { DateTime } from "luxon";
+import { Readable } from "node:stream";
+
+import { BaseScraperModule, TaskHandle } from "#commands/db_scrape";
+import DepartmentModel from "#models/department";
+import DepartmentLinkModel from "#models/department_link";
+import FieldOfStudyModel from "#models/field_of_study";
+import FilesService from "#services/files_service";
+
+interface SourceResponse<T> {
+  data: T[];
+}
+
+interface DepartmentsDraft {
+  id: number;
+  name: string;
+  logo: string | null;
+  description: string | null;
+  code: string;
+  gradient_start: string;
+  gradient_end: string;
+  address: string;
+  betterCode: string;
+  createdAt: string;
+  updatedAt: string | null;
+}
+
+interface FieldOfStudyDraft {
+  id: number;
+  department_id: number;
+  name: string;
+  url: string | null;
+  isEnglish: boolean;
+  is2ndDegree: boolean;
+  isLongCycleStudies: boolean;
+  hasWeekendModeOption: boolean;
+  createdAt: string;
+  updatedAt: string | null;
+}
+
+interface DepartmentLinkDraft {
+  id: number;
+  department_id: number;
+  linkType: string;
+  link: string;
+  createdAt: string;
+  updatedAt: string | null;
+}
+
+export default class DepartmentsScraper extends BaseScraperModule {
+  static name = "Departments";
+  static description =
+    "Scrapes pwr departments, field of study and departments links data from directus";
+  static taskTitle = "Scrape departments, field of study and departments links";
+
+  private filesService = new FilesService();
+
+  private readonly directusSchemas = [
+    "Departments",
+    "FieldOfStudy",
+    "Departments_Links",
+  ];
+  private readonly addressRegex =
+    /\b\d{2}-\d{3}\s+[A-ZĄĆĘŁŃÓŚŹŻa-ząćęłńóśźż-]+$/;
+
+  async run(task: TaskHandle) {
+    task.update("Starting fetching all schema objects");
+
+    const [Departments, FieldOfStudy, DepartmentLink] = (await Promise.all(
```

Comment:
use different names here, don't shadow existing model names
also PascalCase usually implies that this is a class, not an instance - which is not the case here